app/config/AppConfig.scala

```diff
@@ -66,6 +69,16 @@ trait AppConfig {
   def endpointsEnabled(version: Version): Boolean
   def featureSwitches: Configuration
 
+  def apiDocumentationUrl: String
+
+  def isApiDeprecated(version: Version): Boolean = apiStatus(version) == "DEPRECATED"
+
+  def deprecatedOn(version: Version): Option[LocalDateTime]
+
+  def sunsetDate(version: Version): Option[LocalDateTime]
+
+  def isSunsetEnabled(version: Version): Boolean
```

Comment:
Since these values are tightly coupled, it might be a good idea to pull them all together as a single method returning an object that describes the deprecation. This will keep the logic related to what config is allowed with what other config in one place (in AppConfig where it arguably belongs) also reduce the amount of mocking required when a test needs to touch the deprecation related properties.
 
```suggestion 
  def deprecationFor(version: Version): Validated[String, Deprecation]
```

Where `Deprecation` is a trait something like:

```scala
sealed trait Deprecation

object Deprecation {
  case object NotDeprecated extends Deprecation

  case class Deprecated(deprecatedOn: LocalDateTime, sunset: Option[LocalDateTime]) extends Deprecation
}
```
because the version is either deprecated (with a mandatory deprecatedOn and possibly a sunset date time) or it is not. (Ideally all of the API status properties, endpoints enabled flags etc ought to be exposed through one all-encompassing `Status` trait or something, but that might be a step too far at the moment.)